runtime/intern.c

```diff
@@ -596,6 +596,9 @@ static void intern_rec(struct caml_intern_state* s,
         sp->arg = ofs;
         ReadItems(s, dest, 1);
         continue;  /* with next iteration of main loop, skipping *dest = v */
+      case OLD_CODE_CUSTOM:
+        caml_failwith("input_value: custom blocks serialized with "
+                      "OCaml 4.08.0 (or prior) are no longer supported");
```

Comment:
looking at other uses of `caml_failwith` in this file, it looks like you need to call `intern_cleanups(s)` first.

Reply:
It also looks like the house style is to include a `break;` after the failure, probably to avoid a warning.

Reply:
Thanks for the review. I've amended my commit.